### PR TITLE
Optimize shared memory mapping and copy for input and output buffers

### PR DESCRIPTION
**Pull request by huningxin**
To upload an input buffer to GPU uploading buffer
1. Renderer process copies from JS ArrayBuffer to mojo shared memory
2. GPU process copies from shared memory to GPU uploading buffer

To download an output buffer from GPU readback buffer
1. GPU process copies from GPU readback buffer to mojo shared memory
2. Renderer process copies from mojo shared memory to JS ArrayBuffer

This PR reduces the overhead of copying inputs and outputs by:
1. Avoid mapping shared memory for every compute
1. Copy input/output and uploading/readback buffers in one big chunk

It also removes `#pragma optimize("", off)` and adds some trace events for performance profiling.

### CODE REVIEW

File: third_party/blink/renderer/modules/ml/webnn/mojo_graph.cc [hunk 1]
```diff
@@ -279,61 +279,70 @@ void MojoGraph::ComputeAsyncImpl(const MLNamedArrayBufferViews& inputs,
...
     return;
   };
-  for (const auto& output : outputs) {
-    String error_message;
-    void* output_buffer_address = output.second->BaseAddressMaybeShared();
-    if (output_buffer_address == nullptr) {
-      exception_state.ThrowDOMException(DOMExceptionCode::kOperationError,
-                                        error_message);
-      return;
+  {
```
Comment (fdwr) [inline third_party/blink/renderer/modules/ml/webnn/mojo_graph.cc:319]: The only suggestion I'd make is to leave some blank space between the sections to help visually demark them. (but not waiting for it)

**Review by fdwr — APPROVED** [review]
Thanks Ninxgin. I'm unfamiliar with these tracing statements, but they look quite useful from the diagrams you shared in the chat. It appears many separate memory copies were consolidated into just one, which should help stream memory better than multiple fragments (which *might* even be out of order (?) and not spatially local).

I've been meaning to remove the optimization disablement anyway (it's just that debug builds are too unstable, and release builds are difficult to debug without them).

File: third_party/blink/renderer/modules/ml/webnn/mojo_graph.cc (hunk 1 above)
Comment (huningxin) [reply third_party/blink/renderer/modules/ml/webnn/mojo_graph.cc:319]: Good point. I'll follow that in the following PRs. Thanks!

**Comment by huningxin** [comment]
> I've been meaning to remove the optimization disablement anyway (it's just that debug builds are too unstable, and release builds are difficult to debug without them).

Acknowledged. I didn't observe much perf diff after removing these statements in release build. So, they are just some code cleans.

**Pull request merged**